rows/plugins/plugin_csv.py

```diff
@@ -47,6 +47,8 @@ def discover_dialect(sample, encoding):
             # Could not detect dialect, fall back to 'excel'
             return unicodecsv.excel
 
+def get_dialect_parameters(params, dialect, parameter):
+        return params.pop(parameter, getattr(dialect, parameter))
```

Comment:
The reason of kwargs `pop`ing is not messing the future calls that use `kwargs` in the same functions.
Using *args and **kwargs is always a decision on where the code will be messed up.